lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4624,8 +4625,12 @@ namespace {
 
       // While importing the DeclContext, we might have imported the decl
       // itself.
-      if (auto Known = Impl.importDeclCached(decl, getVersion()))
-        return Known;
+      auto Known = Impl.importDeclCached(decl, getVersion());
+      if (Known.hasValue() && Known.getValue()) {
+        return Known.getValue();
+      } else if (Known.hasValue()) {
+        // TODO: Handle case where we have a failed-cached-import here.
+      }
```

Comment:
```suggestion
      auto Known = Impl.importDeclCached(decl, getVersion());
      if (Known.hasValue())
        return Known.getValue();
```

Reply:
Ah ok. I can do this. I was thinking of leaving the TODO blocks to hook up to the diag in a later patch; sorry should have been more clear here.